src/NServiceBus.Core/Pipeline/BehaviorContext.cs

```diff
@@ -7,7 +7,7 @@ namespace NServiceBus
     /// <summary>
     /// Provides base context for behavior context implementations.
     /// </summary>
-    public abstract class BehaviorContext : ContextBag, IBehaviorContext
+    abstract class BehaviorContext : ContextBag, IBehaviorContext
```

Comment:
good question. I'd say no because:
* The only reason to implement this would be to write a custom IBehaviorContext fake for testing purposes. 
* More important: it makes the builder property available which isn't properly configured when not using `RootContext` (which is internal too), so there is a danger of running into exceptions because the custom implementation doesn't setup correctly.